server/src/lexer.rs

```diff
@@ -819,7 +819,207 @@ pub fn source_lexer(
                 }
 
                 RegexDelimType::BlockComment(closing_regex) => {
-                    panic!("Unimplemented.")
+                    let full_comment;
+
+                    // A block comment opening delimiter was found. Find the closing delimiter.
+                    // for now just look for star slash
+                    let end_of_comment_index = source_code.find("*/");
+
+                    // set a boolean variable to indicate no closing delimiter was found
+                    let no_closing_delimiter_found = end_of_comment_index.is_none();
+
+                    // if no closing delimiter was found, then the comment goes to the end of the file and we need to handle it right away, otherwise checking for the closing delimiter will cause the code to hang
+
+                    // case 3: closing delimiter not found
+                    // full_comment should include everything up to the end of the file
+                    // "/* comment " -> "/* comment "
+
+                    // find the FIRST newline AFTER the closing delimiter
+                    let newline_after_comment_index = if no_closing_delimiter_found {
+                        None
+                    } else {
+                        source_code[end_of_comment_index.unwrap() + 2..]
+                            .find('\n')
+                            .map(|i| i + end_of_comment_index.unwrap() + 2)
+                    };
+
+                    //  Assign full_comment to contain the entire comment, from the opening delimiter up to the NEWLINE after the closing delimiter.
```

Comment:
Use the CodeChat Editor to make these into paragraphs, instead of putting blank code lines between them.